engine/sources/local/localfs.go

```diff
@@ -463,8 +464,11 @@ func (local *localFS) toRootPath(path string) string {
 //
 // Unlike other methods below, we don't need to verifyExpectedChange since there was no change applied to the path.
 func (local *localFS) GetPreviousChange(ctx context.Context, path string, stat *types.Stat) (CachedChange, error) {
-	rootPath := local.toRootPath(path)
-	return local.changeCache.GetOrInitialize(ctx, rootPath, func(_ context.Context) (*ChangeWithStat, error) {
+	cacheKey := cache.CacheKey[string]{
+		ResultKey:      local.toRootPath(path),
+		ConcurrencyKey: cacheConcurrencyKey,
+	}
+	return local.changeCache.GetOrInitialize(ctx, cacheKey, func(_ context.Context) (*ChangeWithStat, error) {
```

Comment:
Could maybe do this with a helper, like `cache.ConcurrentCacheKey(local.toRootPath)`.